### DISCUSSION TITLE
Reload binary into existing BNDB?

### DISCUSSION THREAD

**Discussion by alexrp** [Vector35/binaryninja-api#4387] [Q&A] [answered]
If I've done some patching of an executable externally, is there a convenient way to reload it into an existing BNDB?

**Comment by alexrp** [comment]
(ping?)

**Comment by plafosse** [comment, marked as answer]
No there's no way of doing this through the UI. You could pretty easily do this through the API. 

```python
bndb = load("myfile.bndb")
my_patched_file = open("my_patched_file").read()
bndb.parent_view.file.write(my_patched_file)
```
Warning though this won't cause the initial BinaryView parsing to be done again. So if you modified symbol tables or segments/sections then that wouldn't be updated by the above method.

**Reply by alexrp** [reply]
> Warning though this won't cause the initial BinaryView parsing to be done again. So if you modified symbol tables or segments/sections then that wouldn't be updated by the above method.

Is there a way to cause such a re-parse to happen? If not, could support for it be added, or are there significant technical challenges with that?
